src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -75,7 +75,7 @@
      * @param other The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<?> other);
```

Comment:
Then O should be an upper-bound

```
AITask<? super O> other
```

Having ? is worse than O as you'll not be able to infer ANYTHING about it.